struct_amd64.go

```diff
@@ -76,6 +76,17 @@ func tryPlaceRegister(v reflect.Value, addFloat func(uintptr), addInt func(uintp
 	var shift byte // # of bits to shift
 	var flushed bool
 	class := _NO_CLASS
+	flush := func() {
```

Comment:
What about

```go
flushIfNeeded := func() {
    if flushed {
        return
    }
    flushed = true
    // ...
```

so that we don't need an if statement to call flushing?

Reply:
I don't see any benefits in it, could you please help me with it?

`if flushed` is used only once after we've iterated over all fields. It's only purpose is to flush the last small fields, that didn't trigger `if shift == 64` condition. But if we move it to the `flush` function, it'll be executed every flush. Therefore adding a tiny but still accumulating penalty to calls with structures as arguments.

Reply:
This is just for readability and maintenability around callers.

For the penalty, yes, you are right. I was a little worried about the future when we need another flush call but forgetting `if flushed` check, but as long as this unlikely happens, the current change seems fine enough.